rocket/src/FPU.scala

```diff
@@ -301,159 +430,16 @@ object FType {
   val all = List(H, S, D)
 }
 
-trait HasFPUParameters {
```

Comment:
How about merge `HasFPUParameters` and `FPUParams` together? and `FPUModule` is defined with: `FPUModule(param: FPUParams)`.

Reply:
In the latest commit, `FPUModule` is refactored to `HasFPUParameters` trait, so there is no need for a redundant abstract class. Mutable parameters are passed from explicitly class parameters to trait to keep consistency and simplicity. Namely, the original `HasFPUParameters` and `FPUParams` are merged to the same trait `HasFPUParameters` now.